src/tools/rustdoc-gui/tester.js

```diff
@@ -101,25 +115,88 @@ async function main(argv) {
+        files = fs.readdirSync(opts["tests_folder"]);
     } else {
-        files = opts["files"].filter(file => path.extname(file) == ".goml");
+        files = opts["files"];
+    }
+    files = files.filter(file => path.extname(file) == ".goml");
+    if (files.length === 0) {
+        console.log("rustdoc-gui: No test selected");
+        process.exit(2);
     }
-
     files.sort();
-    for (var i = 0; i < files.length; ++i) {
+
+    console.log(`running ${files.length} rustdoc-gui tests`);
+    process.setMaxListeners(files.length + 1);
+    let tests = [];
+    let results = new Array(files.length);
+    // poormans enum
+    const RUN_SUCCESS = 42, RUN_FAILED = 23, RUN_ERRORED = 13;
+    for (let i = 0; i < files.length; ++i) {
         const testPath = path.join(opts["tests_folder"], files[i]);
-        await runTest(testPath, options).then(out => {
-            const [output, nb_failures] = out;
-            console.log(output);
-            if (nb_failures > 0) {
+        tests.push(
+            runTest(testPath, options)
+            .then(out => {
+                //console.log(i);
+                const [output, nb_failures] = out;
+                results[i] = {
+                    status: nb_failures === 0 ? RUN_SUCCESS : RUN_FAILED,
+                    output: output,
+                };
+                if (nb_failures > 0) {
+                    print_test_erroneous()
+                    failed = true;
+                } else {
+                    print_test_successful()
+                }
+            })
+            .catch(err => {
+                results[i] = {
+                    status: RUN_ERRORED,
+                    output: err,
+                };
+                print_test_erroneous();
                 failed = true;
-            }
-        }).catch(err => {
-            console.error(err);
-            failed = true;
-        });
+            })
+        );
+        if (no_headless) {
+            await tests[i];
+        }
     }
+    await Promise.all(tests);
+    // final \n after the tests
+    console.log("\n");
+
+    results.forEach(r => {
+        switch (r.status) {
+            case RUN_SUCCESS:
+                if (debug === false) {
+                    break;
+                }
+            case RUN_FAILED:
+                console.log(r.output);
+                break;
+            case RUN_ERRORED:
+                // skip
+                break;
+            default:
+                console.error(`unexpected status = ${r.status}`);
+                process.exit(4);
+        }
+    });
+    // print run errors on the bottom so developers see them better
+    results.forEach(r => {
+        switch (r.status) {
+            case RUN_SUCCESS:
+            case RUN_FAILED:
+                // skip
+                break;
+            case RUN_ERRORED:
+                console.error(r.output);
+                break;
+        }
```

Comment:
I think you can simply rewrite it as follows:

```suggestion
        if (r.status === RUN_ERRORED) {
            console.error(r.output);
        }
```

Reply:
Oh, yes that is better :+1: